### DISCUSSION TITLE
Serve images from Markdown Documentation as static assets

### DISCUSSION THREAD

**Discussion by judgedim** [structurizr/lite#164] [Q&A]
When linking markdown documentation of a software system that includes images, the documentation is rendered correctly in Structurizr. However, adding even a few images (less than 5MB in total) to a single markdown document makes the UI (both Lite and on-premises) incredibly slow.

![Selection_487](https://github.com/user-attachments/assets/8d323d5a-e541-43bd-8d18-8306dfc70380)

I found here: https://github.com/structurizr/lite/discussions/161#discussioncomment-10616370 that inlined/embedded images is default behavior.

@simonbrowndotje Is it possible to configure Lite and on-premises to provide publicly accessible URLs for each image, instead of converting them to Base64? I mean without host images separately, so web-server from Lite and on-premises will serve them images as static files? 

Thank you in advance for your help!

**Comment by simonbrowndotje** [comment]
There's no way to currently do this, but it can be added for Lite ... feel free to open a feature request.

The on-premises installation is more problematic though, given the additional role-based access, sharing link, etc.

**Reply by judgedim** [reply]
Thanks for the quick reply. Can I ask for some help to get started?

**Comment by simonbrowndotje** [comment]
Moving to #165.
